source/common/runtime/runtime_impl.cc

```diff
@@ -23,6 +23,29 @@ namespace Runtime {
 
 const size_t RandomGeneratorImpl::UUID_LENGTH = 36;
 
+uint64_t RandomGeneratorImpl::random() {
+  static thread_local uint8_t buffered[2048];
+  static thread_local size_t buffered_idx = sizeof(buffered);
+
+  // Prefetch 2048 bytes of randomness. buffered_idx is initialized to sizeof(buffered),
```

Comment:
question: can we prefetch more? Any reason for choosing 2k vs an entire page size (4kb) ?
In other words, does the performance improve if we prefetch more? In which case, it might make sense to have this configurable [though it might be overkill] or cache a much bigger chunk (8kb).

Reply:
I think 2KiB per thread is probably fine but feel free to experiment. Let's definitely not make it configurable right now.

Reply:
@rshriram I used 2KB based on my experiments with UUID, where there were diminishing returns with using more memory.

As for this PR, generation of 1,000,000,000 `uint64_t` number takes:

     prefetch  | time | improvement
    (uint64_t) | (ms) | (%)
    -------------------------------
           128 | 9653 |
           256 | 6930 | 28% faster  <-- this PR right now
           512 | 5571 | 19% faster
          1024 | 4888 | 12% faster
          2048 | 4594 |  6% faster
          4096 | 4424 |  4% faster
          8192 | 4386 |  1% faster

So yes, prefetching more bytes would make this even faster, but keep in mind that this is microbenchmark and the effect on request/s would be negligible (...and this is already 90x faster than what we have now).

Reply:
Sweet! Would you mind sticking this into a comment so that a future Shriram doesn’t come around trying to optimize this?

Reply:
Done.